integration/conftest.py

```diff
@@ -96,28 +91,22 @@ def reactor():
 
 
 @pytest.fixture(scope='session')
-@log_call(action_type=u"integration:temp_dir", include_args=[])
-def temp_dir(request):
+@log_call(action_type=u"integration:base_dir", include_args=[])
+def base_dir(request, tmp_path_factory):
     """
-    Invoke like 'py.test --keep-tempdir ...' to avoid deleting the temp-dir
+    Base temporary directory, used by several session scoped fixtures.
     """
-    tmp = mkdtemp(prefix="tahoe")
-    if request.config.getoption('keep'):
-        print("\nWill retain tempdir '{}'".format(tmp))
-
-    # I'm leaving this in and always calling it so that the tempdir
-    # path is (also) printed out near the end of the run
-    def cleanup():
-        if request.config.getoption('keep'):
-            print("Keeping tempdir '{}'".format(tmp))
-        else:
-            try:
-                shutil.rmtree(tmp, ignore_errors=True)
-            except Exception as e:
-                print("Failed to remove tmpdir: {}".format(e))
-    request.addfinalizer(cleanup)
+    base = tmp_path_factory.mktemp("magic-folder")
+    return str(base)
```

Comment:
Should this really be `str` and not `bytes` or `unicode`?

Reply:
I *think* so yes. I'm specifically wanting to call `pathlib2.Path._str__` here. The docstring for `__bytes__` suggests it should only be used on unix. However, on python2, unicode-mode `FilePath` doesn't allow filesystem operations[1].

[1] This may just be because it forwards the values to the underlying `os` etc function, which tries to encode to ascii.